src/attributes/classes.js

```diff
@@ -14,7 +14,23 @@ function getClass( elem ) {
 
 jQuery.fn.extend( {
 	addClass: function( value ) {
-		var classes, elem, cur, curValue, clazz, j, finalValue,
+		var classes = [];
+		var proceed = false;
+		if ( typeof value === "string" && value ) {
+			classes = value.match( rnothtmlwhite ) || [];
+			proceed = true;
+		} else if ( value.constructor === Array ) {
```

Comment:
I will change it to` Array.isArray`. Why shouldn't I use `value.constructor === Array`